hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveViewCommits.java

```diff
@@ -0,0 +1,442 @@
+    ViewMetadata metadataV1 = ops.current();
+    assertThat(ops.current().properties()).hasSize(0);
+
+    view.updateProperties().set("k0", "v0").commit();
+    ops.refresh();
+    ViewMetadata metadataV2 = ops.current();
+    assertThat(ops.current().properties()).hasSize(1).containsEntry("k0", "v0");
+
+    HiveViewOperations spyOps = spy(ops);
+
+    AtomicReference<HiveLock> lock = new AtomicReference<>();
+    doAnswer(
+            l -> {
+              lock.set(ops.lockObject());
+              return lock.get();
+            })
+        .when(spyOps)
+        .lockObject();
+
+    concurrentCommitAndThrowException(ops, spyOps, (BaseView) view, lock);
+
+    /*
+    This commit should fail and concurrent commit should succeed even though this commit throws an exception
+    after the persist operation succeeds
+     */
+    assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
+        .isInstanceOf(CommitStateUnknownException.class)
+        .hasMessageContaining("Datacenter on fire");
+
+    ops.refresh();
+
+    assertThat(ops.current()).as("Current metadata should have changed").isNotEqualTo(metadataV2);
+    assertThat(metadataFileExists(ops.current()))
+        .as("Current metadata file should still exist")
+        .isTrue();
+    assertThat(ops.current().properties())
+        .as("The new properties from the concurrent commit should have been successful")
+        .hasSize(2);
+  }
+
+  @Test
+  public void testInvalidObjectException() {
+    TableIdentifier badTi = TableIdentifier.of(DB_NAME, "`test_iceberg_view`");
+    assertThatThrownBy(
+            () ->
+                catalog
+                    .buildView(badTi)
+                    .withSchema(SCHEMA)
+                    .withDefaultNamespace(NS)
+                    .withQuery("hive", "select * from ns.tbl")
+                    .create())
+        .isInstanceOf(ValidationException.class)
+        .hasMessage(String.format("Invalid Hive object for %s.%s", DB_NAME, "`test_iceberg_view`"));
+  }
+
+  /** Uses NoLock and pretends we throw an error because of a concurrent commit */
+  @Test
+  public void testNoLockThriftExceptionConcurrentCommit() throws TException, InterruptedException {
+    HiveViewOperations ops = (HiveViewOperations) ((BaseView) view).operations();
+    assertThat(ops.current().properties()).hasSize(0);
+
+    view.updateProperties().set("k1", "v1").commit();
+    ops.refresh();
+    ViewMetadata metadataV2 = ops.current();
+    assertThat(ops.current().properties()).hasSize(1).containsEntry("k1", "v1");
+
+    HiveViewOperations spyOps = spy(ops);
+
+    // Sets NoLock
+    doReturn(new NoLock()).when(spyOps).lockObject();
+
+    // Simulate a concurrent view modification error
+    doThrow(
+            new RuntimeException(
+                "MetaException(message:The table has been modified. The parameter value for key 'metadata_location' is"))
+        .when(spyOps)
+        .persistTable(any(), anyBoolean(), any());
+
```

Comment:
@nk1506 this is missing a `assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))..` check with the expected exception, otherwise what exactly would this test actually verify?

Reply:
more specifically, this should be
```
assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
        .isInstanceOf(CommitFailedException.class)
        .hasMessage("The view hivedb.test_iceberg_view has been modified concurrently");
```

the problem why this exception wasn't thrown is because of the error msg check in `HiveViewOperations`. It needs to be updated to
```
--- a/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveViewOperations.java
+++ b/hive-metastore/src/main/java/org/apache/iceberg/hive/HiveViewOperations.java
@@ -205,7 +205,7 @@ final class HiveViewOperations extends BaseViewOperations implements HiveOperati
         if (e.getMessage() != null
             && e.getMessage()
                 .contains(
-                    "The view has been modified. The parameter value for key '"
+                    "The table has been modified. The parameter value for key '"
```

Reply:
Thanks @nastra for catching this.